src/Module.ts

```diff
@@ -431,11 +433,12 @@ export default class Module {
 		return makeLegal(ext ? base.slice(0, -ext.length) : base);
 	}
 
-	markExports() {
+	markPublicExports() {
 		for (const exportName of this.getExports()) {
 			const variable = this.traceExport(exportName);
 
 			variable.exportName = exportName;
+			variable.reassignPath(UNKNOWN_PATH, NEW_EXECUTION_PATH);
```

Comment:
Just wondering, if we are going to support object truthiness in future like:

```js
var obj = {};
if (!obj || typeof obj !== 'object')
  console.log('treeshaken');
```

Then would it make sense to have a form of `reassignPath` that can treat the object as still an object, but with unknown properties?

So the following optimization could still work:

```js
const obj = {
	reassigned() {},
	test() {
		obj.reassigned();
	}
};

export default obj;

if (!obj)
  console.log('treeshaken');
```

Reply:
This would be supported by the current approach. `UNKNOWN_PATH` might be a sub-optimal variable name as it is actually an "unknown path of length one". Thus the object is still recognized as an object but we no longer have any reliable information about the values of its fields.